Final Legal Chatbot Files/masslegalhelp_legal_tactics_scraper.py

```
@@ -0,0 +1,151 @@
+        driver.get(chapter_url)
+        time.sleep(3)  # wait for page to fully load
+
+        # Find all section headings (h2, h3) that have an id attribute
+        headings = driver.find_elements(By.CSS_SELECTOR, "h2[id], h3[id], h4[id]")
+
+        if not headings:
+            # Fallback: scrape full page as one section
+            print(f"  ⚠️ No anchor headings found — scraping full page for {chapter_name}")
+            try:
+                body = driver.find_element(By.CSS_SELECTOR, "main")
+            except:
```

Ruff (0.15.10)

[error] Do not use bare `except`

(E722)